app/models/lti13.php

```diff
@@ -0,0 +1,530 @@
+<?php
+App::import('Lib', 'Lti13Bootstrap');
+App::import('Lib', 'LTI13Database', array('file'=>'lti13'.DS.'LTI13Database.php'));
+
+use App\LTI13\LTI13Database;
+use Firebase\JWT\JWT;
+use IMSGlobal\LTI\LTI_Exception;
+use IMSGlobal\LTI\LTI_Message_Launch;
+
+/**
+ * LTI 1.3 Model
+ *
+ * @uses      AppModel
+ * @package   CTLT.iPeer
+ * @since     3.4.5
+ * @author    Steven Marshall <steven.marshall@ubc.ca>
+ * @license   MIT {@link http://www.opensource.org/licenses/MIT}
+ * @link      https://www.imsglobal.org/spec/security/v1p0/#fig_oidcflow
+ */
+class Lti13 extends AppModel
+{
+    public $useTable = false;
+    public $db, $User, $Course, $Role, $LtiToolRegistration;
+    public $ltiCourse;
+    public $jwtBody = array();
+    public $ipeerRoster = array();
+    public $ltiRoster = array();
+    public $rosterUpdatesLog = array();
+    public $log_path = ROOT.'/app/tmp/logs/lti13';
+
+    public function __construct()
+    {
+        $this->User = ClassRegistry::init('User');
+        $this->Course = ClassRegistry::init('Course');
+        $this->Role = ClassRegistry::init('Role');
+        $this->LtiToolRegistration = ClassRegistry::init('LtiToolRegistration');
+        $issuers = $this->LtiToolRegistration->findIssuers();
+        $this->db = new LTI13Database($issuers);
+    }
+
+    /**
+     * Initialize LTI_Message_Launch object and validate its data.
+     *
+     * @return string
+     */
+    public function launch()
+    {
+        $launch = LTI_Message_Launch::new($this->db);
+        $launch->validate();
+        return $launch;
+    }
+
+    /**
+     * Get LTI_Message_Launch object from Cache.
+     *
+     * @return LTI_Message_Launch
+     */
+    public function launchFromCache()
+    {
+        $launchId = $this->getLaunchIdFromCache();
+        return LTI_Message_Launch::from_cache($launchId, $this->db);
+    }
+
+    /**
+     * Get launch ID from LTI cache
+     *
+     * @see IMSGlobal\LTI\Cache
+     * @return string
+     */
+    public function getLaunchIdFromCache()
+    {
+        if (!$json = file_get_contents(sys_get_temp_dir() . '/lti_cache.txt')) {
```

Comment:
I'm getting an error oh launching from here
```
Warning (2): file_get_contents(/tmp/lti_cache.txt) [function.file-get-contents]: failed to open stream: No such file or directory [APP/models/lti13.php, line 72]
```

looking into if its something wrong with my setup or something wrong here

Reply:
Okay I figured it out. it was because the `composer.lock` was incorrect. I needed to do a `composer update` to get it working.

please add composer.lock back to the repo and update it 🙏